internal/provider/resource_gitlab_managed_license.go

```diff
@@ -44,11 +48,12 @@ var _ = registerResource("gitlab_managed_license", func() *schema.Resource {
 				Description: "The name of the managed license (I.e., 'Apache License 2.0' or 'MIT license')",
 			},
 			"approval_status": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ForceNew:     false,
-				ValidateFunc: validation.StringInSlice([]string{"approved", "blacklisted"}, true),
-				Description:  "Whether the license is approved or not. Only 'approved' or 'blacklisted' allowed.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         false,
+				ValidateFunc:     validation.StringInSlice(managedLicenseAllowedValues, true),
+				DiffSuppressFunc: checkDeprecatedValuesForDiff,
+				Description:      fmt.Sprintf("The approval status of the license. Valid values are: %s", renderValueListForDocs(managedLicenseAllowedValues)),
```

Comment:
Should we document that the valid values are consisting of deprecated and new values and that some are "treated" as equivalent?

Reply:
Sure - I added a note to the end stating the equivalence and that "allowed" and "denied" should be favored since the others will go away in 15.0. I also added a note about when "allowed" and "denied" were added in the event someone is still using the tf provider on older versions of self-hosted.